numpy/lib/_function_base_impl.py

```diff
@@ -1040,28 +1040,30 @@ def gradient(f, *varargs, axis=None, edge_order=1):
 
     Or a non uniform one:
 
-    >>> x = np.array([0., 1., 1.5, 3.5, 4., 6.], dtype=float)
+    >>> x = np.array([0., 1., 1.5, 3.5, 4., 6.])
     >>> np.gradient(f, x)
     array([1. ,  3. ,  3.5,  6.7,  6.9,  2.5])
 
     For two dimensional arrays, the return will be two arrays ordered by
     axis. In this example the first array stands for the gradient in
     rows and the second one in columns direction:
 
-    >>> np.gradient(np.array([[1, 2, 6], [3, 4, 5]], dtype=float))
-    [array([[ 2.,  2., -1.],
-           [ 2.,  2., -1.]]), array([[1. , 2.5, 4. ],
-           [1. , 1. , 1. ]])]
+    >>> np.gradient(np.array([[1, 2, 6], [3, 4, 5]]))
+    (array([[ 2.,  2., -1.],
+            [ 2.,  2., -1.]]),
+     array([[1. , 2.5, 4. ],
+            [1. , 1. , 1. ]]))
```

Comment:
I'm afraid that you can't just change the formatting of the outputs to be more readable. The outputs need to be exactly what you get from running the inputs. Also note that the returned value is a list, not a tuple.

Reply:
I'm running on `numpy 2.1.0.dev0` and that's the output I'm getting. I can remove this commit from this PR if that's too much to merge.

Reply:
Okay, it's now a tuple (though that might be a bug because the documentation still claims it to be a list), but Python doesn't format tuples like that either.

```python
>>> np.gradient(np.array([[1, 2, 6], [3, 4, 5]]))
(array([[ 2.,  2., -1.],
       [ 2.,  2., -1.]]), array([[1. , 2.5, 4. ],
       [1. , 1. , 1. ]]))
>>> np.__version__
'2.1.0.dev0+git20240526.d1fe3db'
```

You might be using IPython, which will do more pretty-printing.

Reply:
I always copy-paste from IPython, and that is usually fine (unless CI complains, which is rare and it didn't in this case). If it's more readable, it should be fine right?

Reply:
Fair. I didn't realize we went to such lengths beyond vanilla doctest to allow more variability.

Though it does seem that our logic allowed too much and never caught that the return type had changed from list to tuple in the 2.0 transition.

Reply:
There probably isn't an explicit type check for the sequence type when a sequence of arrays is encountered. It's very rare for that to be needed. @ev-br maybe you want to have a look and see if this is useful to add in `scipy-doctest`?

Reply:
In this case, it's a documented type, that was purposefully changed in the 2.0 transition, but now incorrectly documented in both the examples and the `Returns`. We cared enough whether it was a list or a tuple to change it only at the 2.0 transition. AFAICT, it's unintentionally falling through the cracks of the logic chain in our overrides of vanilla doctest meant to handle other cases (e.g. the case of heterogeneous tuples), judging by the comments in there. And because of that, CI never caught the change to indicate that our examples were out of date.

Reply:
But I digress. The whitespace variability that allows IPython pretty-printing is explicit, so the formatting is fine (and now correctly showing the current type). Now just the `Returns` needs to be fixed.

Reply:
Indeed, the doctesting layer never thought about this, and was more concerned about things like `(1, [2, 3, 4])` or `(0, np.arange(3))`. 

Adding a special case for tuples-of-arrays vs lists-of-arrays is definitely doable, so https://github.com/scipy/scipy_doctest/issues/156